securesystemslib/rsa_keys.py

```diff
@@ -246,7 +246,28 @@ def generate_rsa_public_and_private(bits=_DEFAULT_RSA_KEY_BITS):
 
 
 
-def create_rsa_signature(private_key, data, scheme='rsassa-pss-sha256'):
+# The RSA-PSS scheme allows for choosing the salt length:
+# https://crypto.stackexchange.com/questions/1217/rsa-pss-salt-size
+# For compatibility with Golang, we borrow the salt lengths typically used in
+# its crypto package.
+# https://github.com/golang/go/blob/11f92e9dae96939c2d784ae963fa7763c300660b/src/crypto/rsa/pss.go#L225-L232
+# NOTE: the salt length we traditionally used by default is not the default in
+# Golang.
+# FIXME: really, we should encode the salt length as part of the metadata on
+# how to use the RSA-PSS public key.
+class SaltLength:
+	# PSSSaltLengthAuto causes the salt in a PSS signature to be as large
+	# as possible when signing, and to be auto-detected when verifying.
+	PSSSaltLengthAuto = 0
+	# PSSSaltLengthEqualsHash causes the salt length to equal the length
+	# of the hash used in the signature.
+	PSSSaltLengthEqualsHash = -1
+
+
+
+
+
+def create_rsa_signature(private_key, data, scheme='rsassa-pss-sha256', salt_length=SaltLength.PSSSaltLengthEqualsHash):
```

Comment:
Coding style requires we wrap at 79 characters and double-indent the continued line.

Reply:
How did this even pass the test? I thought our travis checks for linewrapping? I made another PR a few days ago and there travis complained about lines longer than 79 characters.. weird

Reply:
No linting on seecuresystemslab yet (#243). Maybe your PR was on in-toto?

Reply:
@lukpueh ah yes.. sorry for the noise :) my mistake